OpenQA/Qemu/Proc.pm

```diff
@@ -405,8 +405,12 @@ sub exec_qemu {
     $process->on(
         collected => sub {
             $self->{_qemu_terminated} = 1;
-            bmwqemu::serialize_state(component => 'backend', msg => 'QEMU exited unexpectedly, see log for details')
-              unless $self->{_stopping};
+            unless ($self->{_stopping}) {
+                my $msg      = 'QEMU exited unexpectedly, see log for details';
+                my $qemu_pid = $self->qemu_pid;
+                $msg = 'QEMU was killed because of out of memory' if (system("dmesg | grep \"Out of memory: Killed process $qemu_pid\"") == 0);
```

Comment:
```suggestion
                $msg = 'QEMU was killed because of out of memory' if qx{dmesg} =~ qr/Out of memory: Killed process $qemu_pid/;
```

This should make https://github.com/os-autoinst/openQA/pull/3903 easier by only needing to allow dmesg, not needing grep

Reply:
Or you just use `index()` as we don't use any actual regex matching here.

Reply:
Nice idea. But that sounds like less clear code and a premature optimization attempt and I don't even know if index is faster than the regex comparison. We only do that in limited cases and on the openQA worker anyway

Reply:
I've searched about it before and the index function is faster than regex. I'm also wondering why `index()` would be less clear. One wouldn't have to worry about escaping which actually sounds like a plus.

Reply:
ok then, regex match as well as index are fine for me

Reply:
Thanks, used `index`.